### DISCUSSION TITLE
Localizations

### DISCUSSION THREAD

**Discussion by tverra** [Daniel-Ioannou/flutter_country_picker#42] [Q&A] [answered]
Hi. I've made an app that uses localizations, as described in the [Flutter docs](https://flutter.dev/docs/development/accessibility-and-localization/internationalization). I've tried to run the example project, and that implementation works. However, when implemented using the official Flutter guide, it doesn't work. I've added a [complete runnable project](https://github.com/Daniel-Ioannou/flutter_country_picker/files/6170079/localizations_test.zip) that demonstrates this issue. 

Other localized strings works as intended , but the `static CountryLocalizations? of(BuildContext context)` method in `country_localizations.dart` returns null, which means that none of the strings in the package are translated. I'm not completely sure what causes this issue, because the inner workings of the Flutter localizations still remains a mystery to me. 

An example of a workaround fould be to return `CountryLocalizations(Localizations.localeOf(context))` instead of `Localizations.of<CountryLocalizations>(context, CountryLocalizations)`, but I don't know if this comes with some kind of cost. Thoughts?

**Comment by Daniel-Ioannou** [comment, marked as answer]
@tverra The package follows the official Flutter guide for localizations. 
For each package that support localization the app must include the package's delegate in the list of the app delegates. So for `country_picker` package you have to add the `CountryLocalizations.delegate` in the list of your app delegates.

If you check the example from the [Flutter docs](https://flutter.dev/docs/development/accessibility-and-localization/internationalization), in this example the app import localizations for Material, Widgets, Cupertino and the app.
 ```Dart
MaterialApp(
 localizationsDelegates: [
   AppLocalizations.delegate,
   GlobalMaterialLocalizations.delegate,
   GlobalWidgetsLocalizations.delegate,
   GlobalCupertinoLocalizations.delegate,
 ],
 supportedLocales: [
    const Locale('en', ''), // English, no country code
    const Locale('ar', ''), // Arabic, no country code
    const Locale.fromSubtags(languageCode: 'zh'), // Chinese *See Advanced Locales below*
  ],
)
```

So in your example project change the code to: 
 ```Dart
MaterialApp(
  title: 'Localizations Demo',
  localizationsDelegates: [
    CountryLocalizations.delegate,
    ...AppLocalizations.localizationsDelegates
  ],
  supportedLocales: AppLocalizations.supportedLocales,
  theme: ThemeData(
    primarySwatch: Colors.blue,
  ),
  home: MyHomePage(),
);
```

**Comment by tverra** [comment]
You're right, I completely missed that. I haven't seen this approach in a Flutter package before. I'm using this package inside my own package, and it kinda seems like just getting the locale from the application context is a more elegant solution than needing the user of my package to add localization delegates for every translatable package their application depends on. But anyway, thanks!
